server/saved_objects/saved_objects_wrapper.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import _ from 'lodash';
+import {
+  HttpServiceStart,
+  SavedObject,
+  SavedObjectsBaseOptions,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkGetObject,
+  SavedObjectsBulkResponse,
+  SavedObjectsBulkUpdateObject,
+  SavedObjectsBulkUpdateOptions,
+  SavedObjectsBulkUpdateResponse,
+  SavedObjectsCheckConflictsObject,
+  SavedObjectsCheckConflictsResponse,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsCreateOptions,
+  SavedObjectsDeleteOptions,
+  SavedObjectsFindOptions,
+  SavedObjectsFindResponse,
+  SavedObjectsUpdateOptions,
+  SavedObjectsUpdateResponse,
+} from 'opensearch-dashboards/server';
+import { OpenSearchDashboardsAuthState } from '../auth/types/authentication_type';
+
+export class SecuritySavedObjectsClientWrapper {
+  public httpStart?: HttpServiceStart;
+
+  constructor() {}
+
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const state: OpenSearchDashboardsAuthState =
+      (this.httpStart!.auth.get(wrapperOptions.request).state as OpenSearchDashboardsAuthState) ||
+      {};
+
+    const selectedTenant = state.selectedTenant;
+    const username = state.authInfo?.user_name;
+    const globalTenant = '';
+    const defaultTenant = 'default';
+    const privateTenant = '__user__';
```

Comment:
use constant? and what if `default` tenant here? Consider building a map between tenant and namespace values to optimize your code here

Reply:
done